dpytools/validation/json/validation.py

```diff
@@ -0,0 +1,91 @@
+import json
+from pathlib import Path
+from typing import Dict, Optional, Union
+from urllib.parse import urlparse
+
+import jsonschema
+from jsonschema import ValidationError
+
+
+def validate_json_schema(
+    schema_path: Union[Path, str],
+    data_dict: Optional[Dict] = None,
+    data_path: Optional[Union[Path, str]] = None,
+    error_msg: Optional[str] = None,
+    indent: Optional[int] = None,
+):
+    """
+    Validate a JSON file against a schema.
+
+    Either `data_dict` or `data_path` must be provided.
+
+    `error_msg` and `indent` can be used to format the error message if validation fails.
+    """
```

Comment:
a general point but you can restructure this to flow a bit better, at the moment its (a) dealing with data path/dict (b) dealing with schema (c) dealing with data path/dict again.

I'd probably deal with the schema stuff first then the rest can just be data dict/path logic then into the validation.

I'd also have a think about assertions, very very useful thing for quickly and cleanly sanitising inputs, eg:

```python
assert not all(data_dict, data_path), "dont provide both dummy"
``` 

which is effectively:

```python
if all([data_path, data_dict)]:
      raise AssertionError("dont provide both dummy")
```

handy thing and maps quite nicely to what you're doing here, you just have to reverse your thinking a bit.